libnetwork/network.go

```diff
@@ -992,6 +993,7 @@ func (n *network) delete(force bool, rmLBEndpoint bool) error {
 	id := n.id
 	n.mu.Unlock()
 
+	c.networkLocker = locker.New()
 	c.networkLocker.Lock(id)
```

Comment:
Doesn't this effectively remove the locking altogether? (i.e., if we create a new locker for every time this is called, effectively, there's no locker?)

Reply:
yes, do you have some idea to  Improve performance for  github.com/moby/locker/locker.go ?

Reply:
> Doesn't this effectively remove the locking altogether? (i.e., if we create a new locker for every time this is called, effectively, there's no locker?)
if I delete c.networkLocker = locker.New(), create and delete will use same lock.

Reply:
Not sure; could be complicated (at least would need digging into cases where these locks would be needed).

It looks like these were originally added in https://github.com/moby/libnetwork/commit/6426d1e66f33c0b0c8bb135b7ee547447f54d043 (part of https://github.com/moby/libnetwork/pull/1808), to address a race-condition, so reverting that would (at least potentially) re-introduce that issue.

Reply:
I try change  form n.ctrlr.networkLocker.Lock(n.id)  to n.ctrlr.networkLocker.Lock(name) maybe solve this problem

Reply:
@mavenugo

Reply:
> this lock is  for avoiding  the race between the network deletion and new endpoints being created.   so it could not been  removed.  
but  originally modification still seems to have problems.   I believe this lock should been move into the caller function of creatpoint.